pkg/cmd/kind/load/docker-image/docker-image.go

```diff
@@ -121,38 +125,42 @@ func runE(logger log.Logger, flags *flagpole, args []string) error {
 
 	// pick only the nodes that don't have the image
 	selectedNodes := []nodes.Node{}
-	for _, node := range candidateNodes {
-		id, err := nodeutils.ImageID(node, imageName)
-		if err != nil || id != imageID {
-			selectedNodes = append(selectedNodes, node)
-			logger.V(0).Infof("Image: %q with ID %q not yet present on node %q, loading...", imageName, imageID, node.String())
+	fns := []func() error{}
+	for i, imageName := range imageNames {
+		imageID := imageIDs[i]
+		for _, node := range candidateNodes {
+			id, err := nodeutils.ImageID(node, imageName)
+			if err != nil || id != imageID {
+				selectedNodes = append(selectedNodes, node)
+				logger.V(0).Infof("Image: %q with ID %q not yet present on node %q, loading...", imageName, imageID, node.String())
+			}
+		}
+		if len(selectedNodes) == 0 {
+			logger.V(0).Infof("Image: %q with ID %q found to be already present on all nodes.", imageName, imageID)
+			continue
 		}
-	}
-
-	if len(selectedNodes) == 0 {
-		return nil
-	}
-
-	// Save the image into a tar
-	dir, err := fs.TempDir("", "image-tar")
-	if err != nil {
-		return errors.Wrap(err, "failed to create tempdir")
-	}
-	defer os.RemoveAll(dir)
-	imageTarPath := filepath.Join(dir, "image.tar")
-
-	err = save(imageName, imageTarPath)
```

Comment:
docker save supports multiple images and dedupes layers.
we should be saving once still.

Reply:
ohh nice. Will do that.